examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java

```diff
@@ -17,30 +15,46 @@
     public static final SimpleDateFormat ADMTimeFormat = new SimpleDateFormat("HH:mm:ss.SSS");
 
     public static String mkADMConstructor(String constructor, String content) {
-        return constructor + "(\"" + content + "\")";
+        StringBuilder sbConstructor=new StringBuilder();
+        sbConstructor.append(constructor).append("(\"").append(content).append("\")");
+        return sbConstructor.toString();
     }
 
     public static String mkInt64Constructor(long value) {
         return mkADMConstructor("int64", String.valueOf(value));
     }
+
     public static String mkInt8Constructor(String value) {
         return mkADMConstructor("int8", value);
     }
+
     public static String mkInt32Constructor(String value) {
         return mkADMConstructor("int32", value);
     }
+
     public static String mkFloatConstructor(String value) {
         return mkADMConstructor("float", (Float.toString(Float.parseFloat(value))) + "f");
     }
-    public static String mkDateConstructor(Date jdate) {
-        return "date(\"" + ADMDateFormat.format(jdate) + "\")";
-    }
-
     public static String mkDateTimeConstructor(Date jdate) {
         return "datetime(\"" + ADMDateFormat.format(jdate) + "T" + ADMTimeFormat.format(jdate) + "\")";
     }
+    public static String mkDateTimeStringFromTweet(String srcDateTimeString) {
+        if(srcDateTimeString.equals(""))
+            return null;
+        StringBuilder sbDateTime=new StringBuilder();
+        sbDateTime.append("parse_datetime(\"").append(srcDateTimeString).append("\",\"W MMM DD hh:mm:ss z YYYY\")");
+        return sbDateTime.toString();
+    }
```

Comment:
Why is StringBuilder necessary? Also, can you put a comment about this function before the function?